api/v1beta1/solrcloud_types.go

```diff
@@ -655,8 +655,8 @@ type SolrCloudStatus struct {
 	// Replicas is the number of number of desired replicas in the cluster
 	Replicas int32 `json:"replicas"`
 
-	// Selector field in the CR status is required by the HPA
-	PodSelector string `json:"selector"`
+	// PodSelector for SolrCloud pods, required by the HPA
+	PodSelector string `json:"podSelector"`
```

Comment:
the field actually has to be called `selector`, according to the [HPA sourcecode](https://github.com/kubernetes/kubernetes/blob/d7355278b34f1eb83ae537506de3a816ac925226/pkg/controller/podautoscaler/horizontal.go#L251)

Reply:
[From the docs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#scale-subresource), I think that `Scale.Status.Selector` is a simulated path that corresponds to the path given in the [labelSelectorPath variable](https://github.com/apache/solr-operator/pull/254/files#diff-5b42b48b24fb46286ce7159c2e5fce00e033136c59267c565e95d4bcfc6301f8R6455).

I'm ok changing this back if it is indeed required to be that exact path, but I would want to make sure first.

Reply:
You're right, I just validated, and here is the result:

```
$ k get --raw "/apis/solr.bloomberg.com/v1beta1/namespaces/my-solr-namespace/solrclouds/my-solr/scale"
...
"status": {
    "replicas": 3,
    "selector": "solr-cloud=my-solr,technology=solr-cloud"
  }
```
The HPA is able to operate too since the `status.field` for `scale` subresource exists and it's correct.
Sorry for the confusion.